plugins/com.google.cloud.tools.eclipse.appengine.libraries/src/com/google/cloud/tools/eclipse/appengine/libraries/config/LibraryBuilder.java

```diff
@@ -0,0 +1,137 @@
+  private static final String ATTR_GROUP_ID = "groupId";
+  private static final String ATTR_REPOSITORY_URI = "repositoryUri";
+  private static final String ATTR_ARTIFACT_ID = "artifactId";
+  private static final String ATTR_VERSION = "version";
+  private static final String ATTR_TYPE = "type";
+  private static final String ATTR_CLASSIFIER = "classifier";
+  private static final String ATTR_EXPORT = "export";
+
+  public Library build(IConfigurationElement configurationElement) throws LibraryBuilderException {
+    try {
+      if (configurationElement.getName().equals(ELMT_LIBRARY)) {
+        Library library = new Library(configurationElement.getAttribute(ATTR_ID));
+        library.setName(configurationElement.getAttribute(ATTR_NAME));
+        library.setSiteUri(new URI(configurationElement.getAttribute(ATTR_SITE_URI)));
+        library.setLibraryFiles(getLibraryFiles(configurationElement.getChildren(ELMT_LIBRARY_FILE)));
+        return library;
+      } else {
+        throw new LibraryBuilderException(MessageFormat.format("Unexpected configuration element with name: {0}. "
+                                                               + "Expected element is {1}",
+                                                               configurationElement.getName(),
+                                                               ELMT_LIBRARY));
+      }
+    } catch (InvalidRegistryObjectException | URISyntaxException exception) {
+      throw new LibraryBuilderException("Error while creating Library instance", exception);
+    }
+  }
+
+  private List<LibraryFile> getLibraryFiles(IConfigurationElement[] children) throws InvalidRegistryObjectException, 
+                                                                                     URISyntaxException {
+    List<LibraryFile> libraryFiles = new ArrayList<>();
+    for (IConfigurationElement libraryFileElement : children) {
+      if (libraryFileElement.getName().equals(ELMT_LIBRARY_FILE)) {
+        MavenCoordinates mavenCoordinates = getMavenCoordinates(libraryFileElement.getChildren(ELMT_MAVEN_COORDINATES));
+        LibraryFile libraryFile = new LibraryFile(mavenCoordinates);
+        libraryFile.setFilters(getFilters(libraryFileElement.getChildren()));
+        libraryFile.setSourceUri(getUri(libraryFileElement.getAttribute(ATTR_SOURCE_URI)));
+        libraryFile.setJavadocUri(getUri(libraryFileElement.getAttribute(ATTR_JAVADOC_URI)));
+        libraryFile.setExport(Boolean.parseBoolean(libraryFileElement.getAttribute(ATTR_EXPORT)));
+        libraryFiles.add(libraryFile);
+      }
+    }
+    return libraryFiles;
+  }
+
+  private URI getUri(String uriString) throws URISyntaxException {
+    if (uriString == null || uriString.isEmpty()) {
+      return null;
+    } else {
+      return new URI(uriString);
+    }
+  }
+
+  private MavenCoordinates getMavenCoordinates(IConfigurationElement[] children) {
+    if (children.length != 1) {
+      // error
```

Comment:
I wouldn't log an error, but a definitely warning.  But I'd move this logic up into `getLibraryFiles()` and just pass in the single `IConfigElement`.  In case we ever need to support multiple jars as a unit?